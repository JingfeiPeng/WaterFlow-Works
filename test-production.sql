-- 1:
-- Query for finding all the prerequisite courses 
-- We can figure out whether this is an All_of group or One_of group 
--   based on whether the rows belong to the same coursegroupid 
SELECT coursecode, coursegroupid
FROM courseGroupMember
WHERE courseGroupID IN (
    SELECT groupid
    FROM courseGroup
    WHERE groupID IN (
        SELECT prereqcoursegroupid
        FROM prerequisite
        WHERE courseCode = UPPER('cs 488')
    )
);

-- 2
-- Query to get the basic information about a course like title, description, etc
SELECT courseCode,title, courseTypes, description, subjectTitle 
FROM course 
WHERE UPPER(courseCode) = UPPER('cs 488');


-- 3
-- Shows how many times a course is in either spring, fall, or winter
SELECT  termCode, COUNT(*) as TimesOffered
FROM (
    SELECT  termCode % 10 as termCode 
    FROM (
        SELECT DISTINCT termCode FROM courseOffering 
        WHERE courseCode = 'MATH 135' AND courseType = 'LEC') As distinctCourses
) as filterTermCode
GROUP BY termCode
ORDER BY termCode DESC;


-- 4
-- Shows when a course is offered in either F,W,S
-- WITH clause is needed since otherwise we can't filter based on term as term is defined in Select
WITH courseTermData AS
(SELECT DISTINCT CASE
    WHEN termCode % 10 = 9 THEN 'F'
    WHEN termCode % 10 = 5 THEN 'S'
    WHEN termCode % 10 = 1 THEN 'W'
END AS Term, 
CASE
    WHEN termCode % 10 = 9 THEN CONCAT(termCode % 1000 / 10 + 2000,'F')
    WHEN termCode % 10 = 5 THEN CONCAT(termCode % 1000 / 10 + 2000,'S')
    WHEN termCode % 10 = 1 THEN CONCAT(termCode % 1000 / 10 + 2000,'W')
END AS Year
FROM courseOffering 
WHERE courseCode = 'CS 350' AND courseType = 'LEC')
SELECT Term, Year
FROM courseTermData WHERE Term = 'F';


-- 5
-- Given a professor shows how many times he/she teaches in Spring, Winter, Fall
SELECT  termCode,profFirstName, profLastName, COUNT(*) as NumOfTimesTaught
FROM (
    SELECT termCode % 10 as termCode, profFirstName, profLastName
    FROM (
        SELECT DISTINCT termCode as termCode, profFirstName, profLastName
        FROM courseOffering
        WHERE courseCode = 'CS 350' AND courseType = 'LEC'
        AND (profFirstName LIKE '%Lesley%'
        OR profLastName LIKE '%Istead%')
    ) as getDistinctTimesTeach
) as tablewithProfCourseTerm
GROUP BY termCode, profFirstName, profLastName
ORDER BY termCode DESC;


-- 6
-- Shows the total history in which a given prof teaches a given course
SELECT  DISTINCT termCode,courseCode, profFirstName, profLastName
FROM courseOffering 
WHERE courseCode = 'CS 350'
AND (profFirstName LIKE '%Lesley%'
OR profLastName LIKE '%Lesley%')
AND courseType = 'LEC'
ORDER BY termCode DESC;


-- 7
-- Shows a List of all the profs that teach a certain course
-- and order them by the number of times they teach
SELECT profFirstName, profLastName, COUNT(*) as numsTaught
FROM courseOffering 
WHERE courseCode = 'CS 350'
AND courseType = 'LEC'
AND profFirstName IS NOT NULL
AND profLastName IS NOT NULL
GROUP BY profFirstName, profLastName
ORDER BY COUNT(*) DESC;



