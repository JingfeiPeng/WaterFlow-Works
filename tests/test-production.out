-> Results for query #1
 coursecode | coursegroupid
------------+---------------
 CS 370     |          3778
 CS 371     |          3778
 CS 350     |          3777
 SE 350     |          3777
(4 rows)

-> Results for query #2
 coursecode |               title               | coursetypes |  description                                                                                                                                                                                                                                                               | subjecttitle                    
------------+-----------------------------------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------
 CS 488     | Introduction to Computer Graphics | LAB,LEC,TST | Software and hardware for interactive computer graphics. Implementation of device drivers, 3-D transformations, clipping, perspective, and input routines. Data structures, hidden surface removal, colour shading techniques, and some additional topics will be covered. | CS
 (1 row)

-> Results for query #3
-> Here 9 = Fall, 5 = Summer and 1 = Winter
 termcode | timesoffered
----------+--------------
        9 |            7
        5 |            7
        1 |            8
(3 rows)

-> Results for query #4
 term | year
------+-------
 F    | 2013F
 F    | 2014F
 F    | 2015F
 F    | 2016F
 F    | 2017F
 F    | 2018F
 F    | 2019F
(7 rows)

-> Results for query #5
 termcode | proffirstname | proflastname | numoftimestaught
----------+---------------+--------------+------------------
        9 | Lesley Ann    | Istead       |                3
        5 | Lesley Ann    | Istead       |                3
        1 | Lesley Ann    | Istead       |                4
(3 rows)

-> Results for query #6
 termcode | coursecode | proffirstname | proflastname
----------+------------+---------------+--------------
     1201 | CS 350     | Lesley Ann    | Istead
     1199 | CS 350     | Lesley Ann    | Istead
     1195 | CS 350     | Lesley Ann    | Istead
     1191 | CS 350     | Lesley Ann    | Istead
     1185 | CS 350     | Lesley Ann    | Istead
     1181 | CS 350     | Lesley Ann    | Istead
     1179 | CS 350     | Lesley Ann    | Istead
     1175 | CS 350     | Lesley Ann    | Istead
     1171 | CS 350     | Lesley Ann    | Istead
     1169 | CS 350     | Lesley Ann    | Istead
(10 rows)


-> Results for query #7
 proffirstname | proflastname | sectionstaught | termstaught
---------------+--------------+----------------+-------------
 Lesley Ann    | Istead       |             20 |          10
 Bernard       | Wong         |             10 |           6
 Kevin         | Lanctot      |              8 |           4
 Timothy       | Brecht       |              6 |           3
 Gregor        | Richards     |              6 |           3
 Kenneth       | Salem        |              5 |           4
 Ali           | Mashtizadeh  |              4 |           2
 Eugene        | Zima         |              3 |           3
 Kamal         | Zille Huma   |              2 |           2
 Igor          | Ivkovic      |              2 |           1
(10 rows)


-> Results for query #8
    groupid	  coursecode	   quantity
---------------+--------------+----------------
      118      |    CS 115    |        1
      118      |    CS 135    |        1
      118      |    CS 145    |        1
      119      |    CS 136    |        1
      119      |    CS 146    |        1
      120      |   MATH 127   |        1
      120      |   MATH 137   |        1
      120      |   MATH 147   |        1
...
(144 rows)